src/ast.js

```diff
@@ -254,22 +253,26 @@ AST.prototype.resolvePrecedence = function(result, parser) {
       }
     }
   } else if (
-    result.kind === "cast" &&
-    result.what &&
-    !result.what.parenthesizedExpression
+    (result.kind === "cast" &&
+      result.what &&
+      !result.what.parenthesizedExpression) ||
+    (result.kind === "silent" &&
+      result.expr &&
+      !result.expr.parenthesizedExpression)
   ) {
     // https://github.com/glayzzle/php-parser/issues/172
-    if (result.what.kind === "bin") {
-      buffer = result.what;
-      result.what = result.what.left;
-      this.swapLocations(result, result, result.what, parser);
+    const subject = result.kind === "cast" ? "what" : "expr";
```

Comment:
could be great to rename `what` property to `expr` on `cast` nodes, that way would be more consistent grammatically speaking - the next release will be stable so it's the last chance to introduce breaking changes.

Reply:
I was hoping you'd suggest that :wink: done!